apps/server/src/modules/common-cartridge/service/common-cartridge-export.service.ts

```diff
@@ -201,40 +202,59 @@ export class CommonCartridgeExportService {
 		});
-
-		await Promise.all(card.elements.map((element) => this.addCardElementToOrganization(element, cardOrganization)));
+		const fileMetadataBufferArray: { id: string; name: string; fileBuffer: Buffer; fileDto: FileDto }[] = [];
+		await Promise.all(card.elements.map((element) => this.downloadAndStoreFiles(element, fileMetadataBufferArray)));
+		await Promise.all(
+			card.elements.map((element) =>
+				this.addCardElementToOrganization(element, cardOrganization, fileMetadataBufferArray)
+			)
+		);
 	}
 
-	private async addCardElementToOrganization(
+	private async downloadAndStoreFiles(
 		element: CardResponseElementsInnerDto,
-		cardOrganization: CommonCartridgeOrganizationNode
+		fileMetadataBufferArray: { id: string; name: string; fileBuffer: Buffer; fileDto: FileDto }[]
 	): Promise<void> {
-		if (RichTextElementResponseDto.isRichTextElement(element)) {
-			const resource = this.mapper.mapRichTextElementToResource(element);
-
-			cardOrganization.addResource(resource);
-		}
-
-		if (LinkElementResponseDto.isLinkElement(element)) {
-			const resource = this.mapper.mapLinkElementToResource(element);
-
-			cardOrganization.addResource(resource);
-		}
-
-		if (FileElementResponseDto.isFileElement(element)) {
+		if (element.type === ContentElementType.FILE) {
 			const filesMetadata = await this.filesMetadataClientAdapter.listFilesOfParent(element.id);
+			for (const fileMetadata of filesMetadata) {
+				const file = await this.filesStorageClientAdapter.download(fileMetadata.id, fileMetadata.name);
+				if (file) {
+					fileMetadataBufferArray.push({
+						id: element.id,
+						name: fileMetadata.name,
+						fileBuffer: file,
+						fileDto: fileMetadata,
+					});
+				}
+			}
+		}
+	}
 
-			await Promise.all(
-				filesMetadata.map(async (fileMetadata) => {
-					const file = await this.filesStorageClientAdapter.download(fileMetadata.id, fileMetadata.name);
-
-					if (file) {
-						const resource = this.mapper.mapFileToResource(fileMetadata, file, element);
+	private addCardElementToOrganization(
+		element: CardResponseElementsInnerDto,
+		cardOrganization: CommonCartridgeOrganizationNode,
+		fileMetadataBufferArray: { id: string; name: string; fileBuffer: Buffer; fileDto: FileDto }[]
+	): void {
+		switch (element.type) {
+			case ContentElementType.RICH_TEXT:
+				const resource = this.mapper.mapRichTextElementToResource(element as RichTextElementResponseDto);
+				cardOrganization.addResource(resource);
+				break;
+			case ContentElementType.LINK:
+				const linkResource = this.mapper.mapLinkElementToResource(element as LinkElementResponseDto);
+				cardOrganization.addResource(linkResource);
+				break;
+			case ContentElementType.FILE:
+				for (const fileMetadata of fileMetadataBufferArray) {
+					const { id, fileBuffer, fileDto } = fileMetadata;
+					if (fileBuffer && element.id === id) {
+						const resource = this.mapper.mapFileToResource(fileDto, fileBuffer, element as FileElementResponseDto);

```

Comment:
Rename it please to fileResource
```suggestion
						const fileResource = this.mapper.mapFileToResource(fileDto, fileBuffer, element as FileElementResponseDto);
```

Reply:
done